client/components/svg-canvas.jsx

```diff
@@ -64,23 +96,70 @@ export default class SVGCanvas extends React.Component {
 
     // Put the new object back in the original location
     this.setState({
-      drawnPaths: [...this.state.drawnPaths.slice(0, drawnPathsIdx), newCurrentPath, ...this.state.drawnPaths.slice(drawnPathsIdx + 1)]
+      elements: [...this.state.elements.slice(0, elementsIdx), newCurrentPath, ...this.state.elements.slice(elementsIdx + 1)]
+    });
+  }
+
+  addUserInputToTextData(userInputKey, elementsIdx) {
+    // Check for valid characters BEFORE calling this method
+    const modifyTextObj = this.state.elements[elementsIdx];
+    let newTextData = modifyTextObj.userInput;
+    if (userInputKey === 'Backspace') {
+      newTextData = newTextData.slice(0, -1);
+    } else {
+      newTextData = newTextData + userInputKey;
+    }
+
+    const newCurrentText = {
+      elementType: modifyTextObj.elementType,
+      elementId: modifyTextObj.elementId,
+      startingPoint: modifyTextObj.startingPoint,
+      userInput: newTextData,
+      fill: modifyTextObj.fill,
+      fontSize: modifyTextObj.fontSize
+    };
+
+    this.setState({
+      elements: [...this.state.elements.slice(0, elementsIdx), newCurrentText, ...this.state.elements.slice(elementsIdx + 1)]
+    });
+  }
+
+  removeElement(elementId) {
+    this.setState({
+      elements: this.state.elements.filter(elementDetail => elementDetail.elementId !== Number(elementId))
     });
   }
 
-  removePath(elementId) {
+  finishTextWriting(idToRemove) {
+    if (idToRemove) {
+      // Check if userInput was blank and remove textbox if so
+      if (this.state.elements.find(element => element.elementId === idToRemove).userInput === '') {
+        this.removeElement(idToRemove);
+      }
+    }
     this.setState({
-      drawnPaths: this.state.drawnPaths.filter(pathDetail => pathDetail.elementId !== Number(elementId))
+      currentElementId: null,
+      isTyping: false
     });
   }
 
+  updateCursorType(currentTool) {
+    // expecting this.props.currentTool
+    if (currentTool === 'pen' || currentTool === 'eraser') {
+      return 'url("./images/circle-cursor.png") 64 64, default';
+    } else if (currentTool === 'text') {
+      return 'text';
+    }
```

Comment:
It may be worth refactoring this if/else ladder to a `switch`. Again, a relatively cosmetic suggestion, but facilitates adding cases easily.

```js
switch (currentTool) {
  case 'pen':
  case 'eraser':
    return 'url("./images/circle-cursor.png") 64 64, default';
  case 'text':
    return 'text';
}
```

Reply:
I never looked into `switch` before, will also be updating code to use this too.